features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/SearchFragment.kt

```diff
@@ -175,6 +177,37 @@ class SearchFragment : Fragment(R.layout.fragment_search) {
         viewModel.allMerchantLocations.observe(viewLifecycleOwner) { merchantLocations ->
             merchantLocationsAdapter.submitList(merchantLocations)
         }
+        viewModel.syncProgressLiveDataFromFlow.observe(viewLifecycleOwner) { syncProgress ->
+
+            when (syncProgress.status) {
+                Status.LOADING -> {
+                    binding.apply {
+                        syncStatus.isVisible = true
+                        progress.isVisible = true
+                        progress.progress = syncProgress.data?.toInt() ?: 0
+                        syncMessage.text = getString(R.string.sync_in_progress_not_complete)
+                    }
+                }
+                Status.SUCCESS -> binding.syncStatus.isVisible = false
+                Status.ERROR -> {
+                    binding.apply {
+                        syncStatus.isVisible = true
+                        syncStatus.setBackgroundResource(R.color.red_300)
+                        progress.isVisible = false
+                        when (syncProgress.exception) {
+                            is FileNotFoundException -> syncMessage.text = getString(R.string.sync_in_progress_file_not_found)
+                            else -> syncMessage.text = getString(R.string.sync_in_progress_error)
+                        }
+                    }
+                }
+                Status.CANCELED -> {
+                    binding.apply {
+                        syncStatus.isVisible = true
+                        syncMessage.text = getString(R.string.sync_in_progress_canceled)
+                    }
+                }
```

Comment:
currently, a canceled status is not tracked from the Sync Worker.  Not sure which exception to look for, but this could be useful on devices that are so slow they have to start over.